dockers/README.md

```diff
@@ -0,0 +1,227 @@
+
+##### :eyes: Example
+```dockerfile
+
+FROM alpine:3.9
+
+RUN set eux; 
+        apk add --no-cache \
+            tini
+
+ENTRYPOINT ["/sbin/tini" , "--"]
+```
+
+## <a name="build"></a> Build Scripts and README
+
+To make life easier when building and pushing our images we like to have an easy-to-use `docker_build.sh` that sits next to each Dockerfile. These scripts should have configurable inputs for the version of tools (samtools, picard, zcall, etc.) being used in the image. Additionally, we like to keep a record of the versions built and being used by writing the images to the accompanying `docker_versions.tsv`.
+
+For first time users of these images it is helpful to have a README which gives a high-level overview of the image.
+
+See the examples for samtools([docker_build](./broad/samtools/docker_build.sh), [docker_versions](./broad/samtools/docker_versions.tsv), [README](./broad/samtools/README.md))
+
+## Formatting
+
+Formatting our Dockerfiles consistenty helps improve readability and eases maintenance headaches down the road. The following are a couple of tenants that we follow when writing our Dockerfiles:
+
+* ARGS, ENV, LABEL in that order
+* Always add versions of tools in the LABEL
+* Single RUN steps
+* Alphabetize package install
+* Clean up package index cache
+* Use ; instead of && for line continuation
+* Logically seperate steps within RUN
+* Four spaces per tab indent
+* Short comments to describe each step
+* tini is always default entrypoint
+
+The following is a good example for our `verify_bam_id` image. This Dockerfile shows how to install packages, install tini and clean up cached index files for a debian base image.
+
+##### :eyes: Example
+```dockerfile
+# Have to use debian based image, many of the installed packages here are not available in Alpine
+FROM us.gcr.io/broad-dsp-gcr-public/base/python:debian
+
+ARG GIT_HASH=c1cba76e979904eb69c31520a0d7f5be63c72253
+
+ENV TERM=xterm-256color \
+    BAMID_URL=https://github.com/Griffan/VerifyBamID/archive \
+    TINI_VERSION=v0.19.0
+
+LABEL MAINTAINER="Broad Institute DSDE <dsde-engineering@broadinstitute.org>" \
+        GIT_HASH=${GIT_HASH}
+
+WORKDIR /usr/gitc
+
+# Install dependencies
+RUN set eux; \
+        apt-get update; \
+        apt-get install -y \ 
+            autoconf \
+            cmake \
+            g++ \
+            gcc \
+            git \
+            libbz2-dev \        
+            libcurl4-openssl-dev \
+            libhts-dev \
+            libssl-dev  \
+            unzip \
+            wget \
+            zlib1g-dev \
+    ; \
+# Install BamID
+    wget ${BAMID_URL}/${GIT_HASH}.zip; \
+    unzip ${GIT_HASH}.zip; \
+    \
+    cd VerifyBamID-${GIT_HASH}; \
+    mkdir build;  \
+    cd build; \
+    CC=$(which gcc) CXX=$(which g++) cmake ..; \
+    \
+    cmake; \
+    make; \
+    make test; \
+    \
+    cd ../../; \
+    mv VerifyBamID-${GIT_HASH}/bin/VerifyBamID .; \
+    rm -rf ${GIT_HASH}.zip VerifyBamID-${GIT_HASH} \
+    ; \
```

Comment:
Out of curiosity, why is this on a separate line?

Reply:
good question!

In a dockerfile there are two ways to do command/line concatenation - && or ;

using && all over the place gets a little cluttered so instead we opt for ; at the end of line except in the case where we are starting a new 'logical block' in our RUN command...in this instance the line with just the bare `; /` is used to separate the all of the stuff associated with installing BamID and the other lines associated with installing tini

functionally it doesn't make a difference but its a nice way to visually break up the long RUN commands

this style is inspired by the official Golang and Ruby Dockerfiles

[Golang Example](https://github.com/docker-library/golang/blob/7f1f587018139127c0dc71bc276ca7b0609847f4/1.17-rc/buster/Dockerfile)

[Ruby Example](https://github.com/docker-library/ruby/blob/761ae37f67acc54d49f085dc4e5a2443a77700e6/3.0/buster/Dockerfile)